src/databricks/labs/ucx/runtime.py

```diff
@@ -138,6 +139,20 @@ def assess_pipelines(cfg: WorkspaceConfig):
     crawler.snapshot()
 
 
+@task("assessment", depends_on=[setup_schema])
```

Comment:
```suggestion
@task("assessment", depends_on=[setup_schema], is_aws=True)
```

Here's the real blocker for this PR: at the moment, this task will always break on Azure and GCP because that API is not supported outside of aws. You have to modify install.py to add tasks only "if task.is_aws and self._ws.config.is_aws". Same for queries. For queries we can use file prefixes.

This won't install AWS related checks on azure at all

Reply:
@nfx I was wondering how this was accomplished!
Just following the azure example which didn't differentiate and should be fixed.

Reply:
Oh wait, you want me to add the cloud differentiator into this PR? That should have been done prior to the  AzureServicePrinciple commit. How about we create that differentiator in a separate PR and merge that in first.

Reply:
Let's move this to https://github.com/databrickslabs/ucx/issues/460

Reply:
#460 replaced by https://github.com/databrickslabs/ucx/pull/610